jwst/straylight/tests/test_straylight.py

```diff
@@ -0,0 +1,72 @@
+"""
+Unit tests for straylight correction
+"""
+
+from jwst.datamodels import IFUImageModel
+from jwst.straylight import StraylightStep
+from jwst.straylight.straylight import correct_mrs_modshepard, shepard_2d_kernel
+from jwst.pipeline.collect_pipeline_cfgs import collect_pipeline_cfgs
+import numpy as np
+import pytest
+
+
+def test_correct_mrs_modshepard():
+    
+    image = IFUImageModel((50,50))
+    image.data = np.ones((50,50)) + 30.0
+
+    slice_map = np.ones((50,50))
+    # create 2 slice gaps
+
+    slice_map[:,10:15] = 0
+    slice_map[:,30:35] = 0
+    image.data[:,10:15] = 0.5
+    image.data[:,30:35] = 0.5
+    roi = 4
+    power = 1
+
+    result = correct_mrs_modshepard(image, slice_map, roi, power)
+    assert type(image) is type(result)
+
+def test_correct_detector():
```

Comment:
IMHO, you could run the step off-line yourself with detector = 'MIRIFULONG', and that would be a sufficient test.  Remember that the unit tests are run every time someone submits a PR (in spacetelescope/jwst), and calling the step takes a lot of time, even if the input is small.  You could have a test such as this for a regression test, but I just don't think it's necessary.